src/Method.php

```diff
@@ -29,27 +29,30 @@
 
     public static function fromName(string $name): Method
     {
-        $method = Method::tryFromName($name);
-        if (is_null($method)) {
-            $enumName = static::class;
-            throw new ValueError("$name is not a valid name for enum \"$enumName\"");
+        $method = self::tryFromName($name);
+
+        if (null === $method) {
+            throw new ValueError(sprintf('%s is not a valid name for enum "%s"', $name, static::class));
```

Comment:
As of php 8.0 throw is an expression, so you could do something like
`null !== $method ? $method : throw new ValueError(sprintf('%s is not a valid name for enum "%s"', $name, static::class)); `
Or even better, move all that to the match in `tryFromName` to default branch.

Reply:
> As of php 8.0 throw is an expression, so you could do something like `null !== $method ? $method : throw new ValueError(sprintf('%s is not a valid name for enum "%s"', $name, static::class));`

While very clever, the cognitive load of that line is a much higher.

> Or even better, move all that to the match in tryFromName to default branch.

That's not an option. `tryFromName` is not allowed to throw, only `fromName` is allowed to throw.

Reply:
Indeed, there is no added value into using the ternary notation.